### PR TITLE
XHR cleanup

### PR DESCRIPTION
**Pull request by jridgewell**
- Moves HTTP method assertions into the `normailzeMethod_` function
- Creates `isArray` and `isObject` utilities.

### CODE REVIEW

**Comment by erwinmombay** [comment]
@jridgewell LGTM

File: src/util.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/**
```
Comment (dvoytenko) [inline src/util.js]: Can we name this file something more specific? "util.js" is too broad. I'd be ok if we just named it "object.js".

Comment (jridgewell) [inline src/util.js]: Are you ok with `types.js`?

Comment (dvoytenko) [inline src/util.js]: Yes, I think `types.js` is good. In general my rule of thumb is this: if you want to name something "util" or "common", first try the name "dump" - if it still works then go ahead :)

**Pull request merged**